engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -2282,7 +2282,7 @@ protected void migrate(final VMInstanceVO vm, final long srcHostId, final Deploy
      * Create the mapping of volumes and storage pools. If the user did not enter a mapping on her/his own, we create one using {@link #getDefaultMappingOfVolumesAndStoragePoolForMigration(VirtualMachineProfile, Host)}.
      * If the user provided a mapping, we use whatever the user has provided (check the method {@link #createMappingVolumeAndStoragePoolEnteredByUser(VirtualMachineProfile, Host, Map)}).
      */
-    private Map<Volume, StoragePool> getPoolListForVolumesForMigration(VirtualMachineProfile profile, Host targetHost, Map<Long, Long> volumeToPool) {
+    protected Map<Volume, StoragePool> getPoolListForVolumesForMigration(VirtualMachineProfile profile, Host targetHost, Map<Long, Long> volumeToPool) {
         if (MapUtils.isEmpty(volumeToPool)) {
```

Comment:
Do we even need this?

        if (MapUtils.isEmpty(volumeToPool)) {
            return getDefaultMappingOfVolumesAndStoragePoolForMigration(profile, targetHost);
        }

Won't the code work just fine without it? It doesn't seem like we need a special case for the mapping being empty when it gets here.

Reply:
Now it will. I will execute the changes then.